boi/mrp.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, osv
+
+class mrp_production(osv.osv):
+
+    _inherit = 'mrp.production'
+
+    def _get_name(self, cr, uid, vals, context=None):
+        product_obj = self.pool.get('product.product')
+        order_obj = self.pool.get('sale.order')
+        if vals.get('active_id', False) and vals.get('active_ids', False):
+            for product in product_obj.browse(cr, uid, vals.get('active_ids')):
+                if product.ref_order_id:
+                    name = self.pool.get('ir.sequence').get(cr, uid, 'mrp.production')
+                    order = order_obj.browse(cr, uid, product.ref_order_id.id, context)
+                    if order.product_tag_id.name == 'BOI':
```

Comment:
For v7, when work with ORM always check the existance of the field field. Ex,
`if order and order.product_tag_id and order.product_tag_id.name == 'BOI':`